### PR TITLE
fixes(#414): change plugins config keys to use - instead of camel case

### LINKED ISSUE #414: ~/.kn/config.yaml should accept and retain camel cased configuration
## In what area(s)?

Kn configuration. Also in docs. 

See PR #411 for discussions

Classifications:

> /kind cleanup

## What version of Knative Client?

```bash
maximilien@maxmini:$ ./kn version
Using kn config file: /home/maximilien/.kn/config.yaml
Version:      v20190920-local-94bb1cf-dirty
Build Date:   2019-09-20 20:38:29
Git Revision: 94bb1cf
Support:
- Serving: v0.8.0  v0.7.1
- API(s):  v1alpha1
```

## What version of Knative Serving running on your cluster?

> 0.8.x

## Expected Behavior

Configuration entries in the `~/.kn/config.yaml` file should retain their case sensitivity. 

## Actual Behavior

Currently they seem to always be caseless, e.g., `pluginsdir: ~/.kn/plugins`

## Steps to Reproduce the Problem

Start without any `~/.kn/config.yaml` and create emty file `touch ~/.kn/config.yaml`.
Run `kn --help`
Your  `~/.kn/config.yaml` will be populated with default configurations that are caseless.

### PR DESCRIPTION
**Pull request by maximilien**
Fixes #414

## Proposed Changes

* adopts a new format for config keys and thereby bypassing the need for case sensitivity in Viper
* changed `lookupPluginsInPath` key to `lookup-plugins` and also same for the persistent flag. The PATH part is implied and can be read from help.

Release Note:

- 🧽 cleans up config keys to match persistent flags

### CODE REVIEW

File: pkg/kn/commands/plugin/list.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func listPlugins(cmd *cobra.Command, flags pluginListFlags) error {
 	if flags.verbose {
 		fmt.Fprintf(out, "The following plugins are available, using options:\n")
 		fmt.Fprintf(out, "  - plugins dir: '%s'%s\n", commands.Cfg.PluginsDir, extraLabelIfPathNotExists(pluginPath))
-		fmt.Fprintf(out, "  - lookup plugins in path: '%t'\n", commands.Cfg.LookupPluginsInPath)
+		fmt.Fprintf(out, "  - lookup plugins: '%t'\n", commands.Cfg.LookupPlugins)
```
Comment (navidshaikh) [inline pkg/kn/commands/plugin/list.go]: since this shows up if verbose, lets keep `lookup plugins in $PATH`?

Comment (rhuss) [inline pkg/kn/commands/plugin/list.go]:
```suggestion
		fmt.Fprintf(out, "  - lookup plugins in $PATH: '%t'\n", commands.Cfg.LookupPlugins)
```

Comment (maximilien) [reply pkg/kn/commands/plugin/list.go]: See @rhuss's comment

Comment (maximilien) [reply pkg/kn/commands/plugin/list.go]: OK

**Pull request merged**